internal/datanode/compaction_executor.go

```diff
@@ -120,22 +126,71 @@ func (c *compactionExecutor) channelValidateForCompaction(vChannelName string) b
 	return !c.dropped.Contain(vChannelName)
 }
 
-func (c *compactionExecutor) stopExecutingtaskByVChannelName(vChannelName string) {
-	c.dropped.Insert(vChannelName)
+func (c *compactionExecutor) discardByDroppedChannel(channel string) {
+	c.dropped.Insert(channel)
```

Comment:
it looks like the `dropped` vchannel is never removed
what if a channel is balanced to another datanode and back? this datanode could never be able to execute compaction anymore

Reply:
dropped will be mark as drop if and only if the collection is dropped.